src/frontend/src/optimizer/optimizer_context.rs

```diff
@@ -122,11 +122,13 @@ impl OptimizerContext {
         self.explain_options.trace
     }
 
+    pub fn is_explain_logical(&self) -> bool {
+        self.explain_options.explain_type == ExplainType::Logical
```

Comment:
This is valid as default is physical. Physical only incurrs a cost in the handler when explain is required..